### PR TITLE
multi: Add tx inputs treasuryspend RPC support.

### PR DESCRIPTION
**Pull request by davecgh**
**This is rebased on top of #2470**.

This adds a new field to the `Vin` and `VinPrevOut` fields of `json/rpc/types` to identify if the input is a treasury spend or not and updates both the marshalling code as well as the relevant rpcserver results accordingly.

In addition, it updates the rpcserver help to document the new field.

At a high level, this means the verbose output of `decoderawtransaction`, `getblock`, `getrawtransaction`, `searchrawtransactions`, and websocket transaction accepted notifications will now identify treasury spend inputs.

The rpcserver version is not bumped since there have been no new releases and release candidates since the previous one.

Finally, a separate commit updates the JSON-RPC API documentation for `decoderawtransaction`, `getrawtransaction`, and `searchrawtransactions` to add the new treasury spend output.

**Example output:**

```
$ dcrctl --testnet getrawtransaction 47de5efe98f073b6f33fbffb1743bacb850fbd8be5723195a744086061f5f0f2 1 | jq '.vin[0]'
{
  "treasuryspend": "4054a36eb67457facef5a075f8ebba5d7a9342dd13b45c6d3dd43b22bead35428603fd805eea93674aa7cd120e46e3fa7a2c563926686909c8eb1b06b40237f3a52103beca9bbd227ca6bb5a58e03a36ba2b52fff09093bd7a50aee1193bccd257fb8ac2",
  "sequence": 4294967295,
  "amountin": 1.0000255,
  "blockheight": 0,
  "blockindex": 4294967295
}
```

### CODE REVIEW

**Review by chappjc — APPROVED** [review]
Looks good too.  Haven't had a chance to test it though.

File: internal/rpcserver/rpcserver.go [hunk 1]
```diff
@@ -4163,6 +4175,24 @@ func createVinListPrevOut(s *Server, mtx *wire.MsgTx,
 		return vinList, nil
 	}
 
+	// Treasury spend transactions only have a single txin by definition.
+	if isTreasuryEnabled && stake.IsTSpend(mtx) {
+		// Only include the transaction if the filter map is empty
+		// because a coinbase input has no addresses and so would never
```
Comment (davecgh) [inline internal/rpcserver/rpcserver.go:4181]: s/coinbase/treasury spend/

File: docs/json_rpc_api.mediawiki [hunk 2]
```diff
@@ -618,6 +618,16 @@ the method name for further details such as parameter and return information.
 
 : <code>{"treasurybase": bool, "amountin": n.nnn, "blockheight": n, "blockindex": n, "sequence": n, ...}</code>
 
+; vin (for treasuryspend transactions)
```
Comment (davecgh) [inline docs/json_rpc_api.mediawiki:621]: treasury spend

File: docs/json_rpc_api.mediawiki [hunk 3]
```diff
@@ -1816,6 +1829,16 @@ of the best block.
...
```
Comment (davecgh) [inline docs/json_rpc_api.mediawiki:1832]: treasury spend

**Review by dnldd — APPROVED** [review]

**Review by jrick — APPROVED** [review]

**Pull request merged**